components/styled/table/src/molecules/questionnaire-table.tsx

```diff
@@ -23,24 +25,76 @@ export const mapQuestionnaireObjectsToVerticalTableData = (
       accessor: record?.id ?? '',
     })),
   ],
-  rows: definitionItems.map((def) => ({
-    id: def?.linkId ?? '',
-    cells: [
-      {
-        key: 'property',
-        value: def?.text ?? '',
-      },
-      ...records.map((record) => ({
-        key: record.id,
-        value: EnsureMaybe<string>(
-          record.item?.find((item) => item?.linkId === def?.linkId)?.answer?.find((answer) => !!answer)?.valueString,
-          ''
-        ),
-      })),
-    ],
-  })),
+  rows: buildVerticalCellRowsRecursive(definitionItems, records),
 })
 
+const buildVerticalCellRowsRecursive = (
+  definitionItems: QuestionnaireItem[],
+  records: QuestionnaireResponse[]
+): CellRow[] =>
+  definitionItems.map((def) => {
+    const containsSubRows: boolean = (def?.item && def.item.length > 0) ?? false
+    const subRows = containsSubRows
+      ? buildVerticalCellRowsRecursive(EnsureMaybe(def?.item?.map((x) => EnsureMaybe(x))), records)
+      : []
+
+    return {
+      id: def?.linkId ?? '',
+      cells: [
+        {
+          key: 'property',
+          value: def?.text ?? '',
+        },
+        ...records.map((record) => ({
+          key: record.id,
+          value: containsSubRows ? (
+            <Checkbox
+              size="small"
+              color="primary"
+              checked={subRows.some((x) => x.cells.find((y) => y.key === record.id)?.value !== '')}
+              style={{ padding: 0 }}
+            />
+          ) : (
+            findQuestionnaireResponseAnswerValue(EnsureMaybe(def?.linkId), record?.item ?? [])
```

Comment:
Just checking - are you aware that EnsureMaybe will throw an error if `def?.linkId` is ever undefined? 
If so it might be nice to catch the error and render an error message. 
If not then you can supply a default value as the second argument which it'll fall back to instead of erroring